cli/args/mod.rs

```diff
@@ -1564,7 +1564,99 @@ impl CliOptions {
 
     let mut permissions_options = flags_to_options(&self.flags.permissions);
     self.augment_import_permissions(&mut permissions_options);
-    permissions_options
+
+    let sets = self.start_dir.to_permission_sets()?;
+
+    fn do_thing(
+      allow_deny: &deno_config::deno_json::AllowDeny,
+      current_allow: &mut Option<Vec<String>>,
+      current_deny: &mut Option<Vec<String>>,
+    ) {
+      if current_allow.is_none() {
+        match &allow_deny.allow {
+          deno_config::deno_json::Permission::All => {
+            *current_allow = Some(vec![])
+          }
+          deno_config::deno_json::Permission::Some(vec) => {
+            *current_allow = Some(vec.clone())
+          }
+          _ => {}
+        }
+      }
+      if current_deny.is_none() {
+        match &allow_deny.deny {
+          deno_config::deno_json::Permission::All => {
+            *current_deny = Some(vec![])
+          }
+          deno_config::deno_json::Permission::Some(vec) => {
+            *current_deny = Some(vec.clone())
+          }
+          _ => {}
+        }
+      }
+    }
+
+    macro_rules! merge {
+        ($perms: ident; $($field: ident),*) => {
+            $(
+              paste::paste! {
+                do_thing(&$perms.$field, &mut permissions_options.[<allow_ $field>], &mut permissions_options.[<deny_ $field>]);
+              }
+            )*
+        };
+    }
+
+    macro_rules! set_allows {
+        ($($field: ident),*) => {
+            paste::paste! {
+              $(
+                permissions_options.[<allow_ $field>] = Some(Default::default());
+              )*
+            }
+        };
+    }
+
+    if let Some(name) = &self.flags.permission_set {
+      let set = sets.sets.get(name).ok_or_else(|| {
+        deno_core::anyhow::anyhow!(
+          "failed to find permission set named \"{name}\""
+        )
+      })?;
```

Comment:
I think we shouldn't error if someone does `-P` and the default set doesn't exist. That way someone can run something like `deno test -P` unconditionally.